client/app/components/queries/query-editor.js

```diff
@@ -1,160 +1,6 @@
-                  snippetText: '',
-                };
-                m.snippets = snippetManager.parseSnippetFile(m.snippetText);
-                snippets.forEach((snippet) => {
-                  m.snippets.push(snippet.getSnippet());
-                });
-
-                snippetManager.register(m.snippets || [], m.scope);
-              });
-            });
-
-            editor.$blockScrolling = Infinity;
-            editor.getSession().setUseWrapMode(true);
-            editor.setShowPrintMargin(false);
-
-            $scope.$watch('syntax', (syntax) => {
-              const newMode = `ace/mode/${syntax}`;
-              editor.getSession().setMode(newMode);
-            });
-
-            $scope.$watch('schema', (newSchema, oldSchema) => {
-              if (newSchema !== oldSchema) {
-                if (newSchema === undefined) {
-                  return;
-                }
-                const tokensCount = newSchema.reduce((totalLength, table) => totalLength + table.columns.length, 0);
-                // If there are too many tokens we disable live autocomplete,
-                // as it makes typing slower.
-                if (tokensCount > 5000) {
-                  editor.setOption('enableLiveAutocompletion', false);
-                } else {
-                  editor.setOption('enableLiveAutocompletion', true);
-                }
-              }
-            });
-
-            $scope.$parent.$on('angular-resizable.resizing', () => {
-              editor.resize();
-            });
-
-            editor.focus();
-          },
-        };
-
-        const schemaCompleter = {
-          getCompletions(state, session, pos, prefix, callback) {
-            if (prefix.length === 0 || !$scope.schema) {
-              callback(null, []);
-              return;
-            }
-
-            if (!$scope.schema.keywords) {
-              const keywords = {};
-
-              $scope.schema.forEach((table) => {
-                keywords[table.name] = 'Table';
-
-                table.columns.forEach((c) => {
-                  keywords[c] = 'Column';
-                  keywords[`${table.name}.${c}`] = 'Column';
-                });
-              });
-
-              $scope.schema.keywords = map(keywords, (v, k) => ({
-                name: k,
-                value: k,
-                score: 0,
-                meta: v,
-              }));
-            }
-            callback(null, $scope.schema.keywords);
-          },
-        };
-
-        window.ace.acequire(['ace/ext/language_tools'], (langTools) => {
-          langTools.addCompleter(schemaCompleter);
-        });
-      },
-    },
-  };
-}
+import { react2angular } from 'react2angular';
+import QueryEditor from '@/react-components/QueryEditor';
 
 export default function init(ngModule) {
-  ngModule.directive('queryEditor', queryEditor);
+  ngModule.component('queryEditor', react2angular(QueryEditor, null, ['QuerySnippet', 'Query', 'KeyboardShortcuts']));
```

Comment:
Why not include the `react2angular` declaration with the component itself?

Reply:
Oh, I see what you mean now.